### DISCUSSION TITLE
custom search

### DISCUSSION THREAD

**Discussion by mikitu** [doug-martin/nestjs-query#1253] [General]
Is there a way to pass a custom criteria to the query service ? 
For eg, this search condition, while keeping the default pagination provided by nests-query: 

```
const criteria = {
      where: Sequelize.where(
        Sequelize.fn(
          'concat',
          Sequelize.col('first_name'),
          Sequelize.col('last_name'),
        ),
        {
          like: `%${q.filter.name}%`,
        },
      ),
    };
```

Thank you

**Comment by awHamer** [comment]
Currently, there is no way to do a custom columns like that and search by them which is sad :(
As for your question you can create a generated column in mysql/postgresql for calculating full_name and treat this column as a real one in Sequelize/typeorm.
